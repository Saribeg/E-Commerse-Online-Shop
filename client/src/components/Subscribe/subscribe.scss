@import '../../scss/resources/resources';

.section-subscribe {
  padding: 65px 0;
  position: relative;
}

.subscribe {
  @include flex-center-between;

  @include max-width($landscape-tablet) {
    flex-direction: column;
  }

  &-title {
    font-family: $font-openSans;
    font-size: 38px;
    font-weight: 600;
    color: #31363d;
    margin-right: 24px;

    @include max-width($landscape-tablet) {
      margin-right: 0;
      margin-bottom: 10px;
    }
  }

  &-description {
    width: 243px;
    font-family: $font-openSans;
    font-size: 14px;
    line-height: 1.36;
    color: #31363d;

    @include max-width($landscape-tablet) {
      width: 100%;
      margin-bottom: 20px;
      text-align: center;
    }
    @include min-width(1000px) {
      width: auto;
    }
  }

  &-form {
    display: flex;
    max-width: 658px;
    width: 100%;

    @include max-width($landscape-tablet) {
      flex-direction: column;
      margin-bottom: 10px;
      justify-content: center;
      align-items: center;
    }
  }

  &-email {
    border: none;
    outline: none;
    max-width: 300px;
    width: 100%;
    border-radius: 22px;
    padding: 15px;
    background-color: $grey-light;
    margin-right: 16px;

    @include max-width($landscape-tablet) {
      margin-right: 0;
      margin-bottom: 10px;
    }
  }

  &-btn {
    max-width: 300px;
    width: 100%;
    padding: 12px 0;
    border-radius: 100px;
    background-color: $black;
    font-family: $font-openSans;
    font-size: 13px;
    font-weight: bold;
    text-align: center;
    letter-spacing: 1.4px;
    text-transform: uppercase;
    color: #fff;
    border: none;
    outline: none;
    cursor: pointer;
  }
}

.hintMessage {
  position: absolute;
  bottom: 40px;
  left: 50%;
  //transform: translate(-50%);
  color: red;
}

.sendMail {
  position: absolute;
  bottom: 40px;
  left: 50%;
  //transform: translate(-50%);
  color: green;
}