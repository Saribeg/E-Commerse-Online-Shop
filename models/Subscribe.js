const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const SubscribeSchema = new Schema({
    status: {
        type: Boolean,
        required: true,
        default: true
    },
    mail: {
        type: String,
        required: true
    },
    date: {
        type: Date,
        default: Date.now
    }
});

module.exports = Subscribe = mongoose.model("subscribe", SubscribeSchema);
