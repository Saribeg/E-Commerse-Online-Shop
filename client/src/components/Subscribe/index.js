// @flow

import * as React from "react";
import "./subscribe.scss";
import axios from "axios";
import {requiredInput, correctEmail} from '../../validation'

type Props = {};

type State = {
    mail: string,
    hintMessage: string
};

class Subscribe extends React.Component<Props, State> {
    state = {
        mail: "",
        hintMessage: ""
    };

    handleChange = (e: Object) => {
        this.setState({mail: e.target.value});
    };

    sendMail = (e: Object) => {
        e.preventDefault();
        const subMail = {
            mail: this.state.mail,
        };
        axios.post("/subscribe", {subMail})
            .then(res => res.data)
            .then(data => {
                if (data.result === false) {
                    this.setState({hintMessage: 'alreadyUsed'});
                    setTimeout(() => {
                        this.setState({hintMessage: 'clear'})
                    }, 2000)
                } else {
                    this.setState({mail: "", hintMessage: 'sendMail'});
                    setTimeout(() => {
                        this.setState({hintMessage: 'clear'})
                    }, 2000)
                }
            });

        if (this.state.mail === '') {
            this.setState({hintMessage: 'emptyMail'});
            setTimeout(() => {
                this.setState({hintMessage: 'clear'})
            }, 2000)
        }
    };

    render() {
        let hintMessage;

        if (this.state.hintMessage === 'clear') {
            hintMessage = '';
        } else if (this.state.hintMessage === 'emptyMail') {
            hintMessage = 'Fill in the input field please!';
        } else if (this.state.hintMessage === 'alreadyUsed') {
            hintMessage = 'You are already subscribed.';
        } else if (this.state.hintMessage === 'sendMail') {
            hintMessage = 'Email has been sent!';
        }

        return (
            <>
                <section className="section-subscribe">
                    <div className="container">
                        <div className="subscribe">
                            <h2 className="subscribe-title">Subscribe</h2>
                            <div className="subscribe-description">
                                Get the dayâ€™s top news stories delivered to your inbox
                            </div>
                            <form onSubmit={this.sendMail} className="subscribe-form">
                                <input
                                    onChange={this.handleChange}
                                    id="subscribe-mail"
                                    type="text"
                                    className="subscribe-email"
                                    placeholder="Enter your email here..."
                                    value={this.state.mail}
                                />
                                <input
                                    id="subscribe-btn"
                                    type="submit"
                                    className="subscribe-btn"
                                    value="SUBSCRIBE"
                                />
                            </form>
                        </div>
                        <p className={this.state.hintMessage === 'sendMail' ? 'sendMail' : 'hintMessage'}>{hintMessage}</p>
                    </div>
                </section>
            </>
        );
    }
}

export default Subscribe;
