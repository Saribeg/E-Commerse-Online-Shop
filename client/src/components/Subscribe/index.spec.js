import React from 'react';
import Subscribe from './index';

describe('Subscribe component', () => {
    it('Subscribe component render', () => {
        const wrapper  = shallow(<Subscribe />)
        expect(wrapper).toMatchSnapshot()
    })
})