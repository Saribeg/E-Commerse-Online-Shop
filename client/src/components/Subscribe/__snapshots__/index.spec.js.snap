// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscribe component Subscribe component render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Subscribe />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <section
        className="section-subscribe"
      >
        <div
          className="container"
        >
          <div
            className="subscribe"
          >
            <h2
              className="subscribe-title"
            >
              Subscribe
            </h2>
            <div
              className="subscribe-description"
            >
              Get the day’s top news stories delivered to your inbox
            </div>
            <form
              className="subscribe-form"
              onSubmit={[Function]}
            >
              <input
                className="subscribe-email"
                id="subscribe-mail"
                onChange={[Function]}
                placeholder="Enter your email here..."
                type="text"
                value=""
              />
              <input
                className="subscribe-btn"
                id="subscribe-btn"
                type="submit"
                value="SUBSCRIBE"
              />
            </form>
          </div>
          <p
            className="hintMessage"
          />
        </div>
      </section>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="subscribe"
          >
            <h2
              className="subscribe-title"
            >
              Subscribe
            </h2>
            <div
              className="subscribe-description"
            >
              Get the day’s top news stories delivered to your inbox
            </div>
            <form
              className="subscribe-form"
              onSubmit={[Function]}
            >
              <input
                className="subscribe-email"
                id="subscribe-mail"
                onChange={[Function]}
                placeholder="Enter your email here..."
                type="text"
                value=""
              />
              <input
                className="subscribe-btn"
                id="subscribe-btn"
                type="submit"
                value="SUBSCRIBE"
              />
            </form>
          </div>
          <p
            className="hintMessage"
          />
        </div>,
        "className": "section-subscribe",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="subscribe"
            >
              <h2
                className="subscribe-title"
              >
                Subscribe
              </h2>
              <div
                className="subscribe-description"
              >
                Get the day’s top news stories delivered to your inbox
              </div>
              <form
                className="subscribe-form"
                onSubmit={[Function]}
              >
                <input
                  className="subscribe-email"
                  id="subscribe-mail"
                  onChange={[Function]}
                  placeholder="Enter your email here..."
                  type="text"
                  value=""
                />
                <input
                  className="subscribe-btn"
                  id="subscribe-btn"
                  type="submit"
                  value="SUBSCRIBE"
                />
              </form>
            </div>,
            <p
              className="hintMessage"
            />,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="subscribe-title"
                >
                  Subscribe
                </h2>,
                <div
                  className="subscribe-description"
                >
                  Get the day’s top news stories delivered to your inbox
                </div>,
                <form
                  className="subscribe-form"
                  onSubmit={[Function]}
                >
                  <input
                    className="subscribe-email"
                    id="subscribe-mail"
                    onChange={[Function]}
                    placeholder="Enter your email here..."
                    type="text"
                    value=""
                  />
                  <input
                    className="subscribe-btn"
                    id="subscribe-btn"
                    type="submit"
                    value="SUBSCRIBE"
                  />
                </form>,
              ],
              "className": "subscribe",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Subscribe",
                  "className": "subscribe-title",
                },
                "ref": null,
                "rendered": "Subscribe",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Get the day’s top news stories delivered to your inbox",
                  "className": "subscribe-description",
                },
                "ref": null,
                "rendered": "Get the day’s top news stories delivered to your inbox",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <input
                      className="subscribe-email"
                      id="subscribe-mail"
                      onChange={[Function]}
                      placeholder="Enter your email here..."
                      type="text"
                      value=""
                    />,
                    <input
                      className="subscribe-btn"
                      id="subscribe-btn"
                      type="submit"
                      value="SUBSCRIBE"
                    />,
                  ],
                  "className": "subscribe-form",
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "subscribe-email",
                      "id": "subscribe-mail",
                      "onChange": [Function],
                      "placeholder": "Enter your email here...",
                      "type": "text",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "subscribe-btn",
                      "id": "subscribe-btn",
                      "type": "submit",
                      "value": "SUBSCRIBE",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "form",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "hintMessage",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <section
          className="section-subscribe"
        >
          <div
            className="container"
          >
            <div
              className="subscribe"
            >
              <h2
                className="subscribe-title"
              >
                Subscribe
              </h2>
              <div
                className="subscribe-description"
              >
                Get the day’s top news stories delivered to your inbox
              </div>
              <form
                className="subscribe-form"
                onSubmit={[Function]}
              >
                <input
                  className="subscribe-email"
                  id="subscribe-mail"
                  onChange={[Function]}
                  placeholder="Enter your email here..."
                  type="text"
                  value=""
                />
                <input
                  className="subscribe-btn"
                  id="subscribe-btn"
                  type="submit"
                  value="SUBSCRIBE"
                />
              </form>
            </div>
            <p
              className="hintMessage"
            />
          </div>
        </section>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <div
              className="subscribe"
            >
              <h2
                className="subscribe-title"
              >
                Subscribe
              </h2>
              <div
                className="subscribe-description"
              >
                Get the day’s top news stories delivered to your inbox
              </div>
              <form
                className="subscribe-form"
                onSubmit={[Function]}
              >
                <input
                  className="subscribe-email"
                  id="subscribe-mail"
                  onChange={[Function]}
                  placeholder="Enter your email here..."
                  type="text"
                  value=""
                />
                <input
                  className="subscribe-btn"
                  id="subscribe-btn"
                  type="submit"
                  value="SUBSCRIBE"
                />
              </form>
            </div>
            <p
              className="hintMessage"
            />
          </div>,
          "className": "section-subscribe",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="subscribe"
              >
                <h2
                  className="subscribe-title"
                >
                  Subscribe
                </h2>
                <div
                  className="subscribe-description"
                >
                  Get the day’s top news stories delivered to your inbox
                </div>
                <form
                  className="subscribe-form"
                  onSubmit={[Function]}
                >
                  <input
                    className="subscribe-email"
                    id="subscribe-mail"
                    onChange={[Function]}
                    placeholder="Enter your email here..."
                    type="text"
                    value=""
                  />
                  <input
                    className="subscribe-btn"
                    id="subscribe-btn"
                    type="submit"
                    value="SUBSCRIBE"
                  />
                </form>
              </div>,
              <p
                className="hintMessage"
              />,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2
                    className="subscribe-title"
                  >
                    Subscribe
                  </h2>,
                  <div
                    className="subscribe-description"
                  >
                    Get the day’s top news stories delivered to your inbox
                  </div>,
                  <form
                    className="subscribe-form"
                    onSubmit={[Function]}
                  >
                    <input
                      className="subscribe-email"
                      id="subscribe-mail"
                      onChange={[Function]}
                      placeholder="Enter your email here..."
                      type="text"
                      value=""
                    />
                    <input
                      className="subscribe-btn"
                      id="subscribe-btn"
                      type="submit"
                      value="SUBSCRIBE"
                    />
                  </form>,
                ],
                "className": "subscribe",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Subscribe",
                    "className": "subscribe-title",
                  },
                  "ref": null,
                  "rendered": "Subscribe",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Get the day’s top news stories delivered to your inbox",
                    "className": "subscribe-description",
                  },
                  "ref": null,
                  "rendered": "Get the day’s top news stories delivered to your inbox",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        className="subscribe-email"
                        id="subscribe-mail"
                        onChange={[Function]}
                        placeholder="Enter your email here..."
                        type="text"
                        value=""
                      />,
                      <input
                        className="subscribe-btn"
                        id="subscribe-btn"
                        type="submit"
                        value="SUBSCRIBE"
                      />,
                    ],
                    "className": "subscribe-form",
                    "onSubmit": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "subscribe-email",
                        "id": "subscribe-mail",
                        "onChange": [Function],
                        "placeholder": "Enter your email here...",
                        "type": "text",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "subscribe-btn",
                        "id": "subscribe-btn",
                        "type": "submit",
                        "value": "SUBSCRIBE",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                  ],
                  "type": "form",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "hintMessage",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "section",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
